crates/cdk-axum/src/router_handlers.rs

```diff
@@ -11,9 +11,38 @@ use cdk::nuts::{
     SwapRequest, SwapResponse,
 };
 use cdk::util::unix_time;
+use paste::paste;
 
 use crate::MintState;
 
+macro_rules! post_cache_wrapper {
+    ($handler:ident, $request_type:ty, $response_type:ty) => {
+        paste! {
+            /// Cache wrapper function for $handler:
+            /// Wrap $handler into a function that caches responses using the request as key
+            pub async fn [<cache_ $handler>](
+                state: State<MintState>,
+                payload: Json<$request_type>
+            ) -> Result<Json<$response_type>, Response> {
+                let Json(json_extracted_payload) = payload.clone();
+                let State(mint_state) = state.clone();
+                let cache_key = serde_json::to_string(&json_extracted_payload).unwrap();
+                if let Some(cached_response) = mint_state.cache.get(&cache_key) {
+                    return Ok(Json(serde_json::from_str(&cached_response).unwrap()));
+                }
+
+                let Json(response) = $handler(state, payload).await?;
+                mint_state.cache.insert(cache_key, serde_json::to_string(&response).unwrap()).await;
+                Ok(Json(response))
```

Comment:
Can we remove the use of `unwrap` and return an error response. The use of `unwrap` should be avoided since it will panic at runtime. In cases where we are sure that it won't panic we can use `expect("Reason why were know it wont panic")`. But I think here since we already are returning a response we can return a error response,